master/internal/telemetry/telemetry.go

```diff
@@ -58,35 +62,93 @@ func New(
 			"master_version": version.Version,
 		},
 	}); err != nil {
+		// TODO CAROLINA: should this panic/return nil ?
 		logrus.WithError(err).Warnf("failed to enqueue identity %s", clusterID)
 	}
 
-	return &TelemetryActor{db, rm, client, clusterID}, nil
+	syslog := logrus.WithFields(logrus.Fields{
+		"component":  fmt.Sprintf("telemetry-actor"),
+		"clusterID":  clusterID,
+		"segmentKey": segmentKey,
+	})
+
+	return &TelemetryActor{
+		db:        db,
+		rm:        rm,
+		client:    client,
+		clusterID: clusterID,
+		syslog:    syslog,
+	}, nil
 }
 
-// Receive implements the actor.Actor interface.
-func (s *TelemetryActor) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		actors.NotifyAfter(ctx, 0, telemetryTick{})
+// InitTelemetry sets up the actor for the telemetry.
+func InitTelemetry(
+	system *actor.System,
+	db db.DB,
+	rm telemetryRPFetcher,
+	clusterID string,
+	conf config.TelemetryConfig,
+) {
+	if DefaultTelemetry != nil {
+		logrus.Warn(`detected re-initialization of Telemetry actor `,
+			`that should never occur outside of tests`)
+	}
 
-	case analytics.Track:
-		msg.UserId = s.clusterID
-		if err := s.client.Enqueue(msg); err != nil {
-			ctx.Log().WithError(err).Warnf("failed to enqueue track %s", msg.Event)
+	if conf.Enabled && conf.SegmentMasterKey != "" {
+		if actorDef, tErr := New(
+			db,
+			rm,
+			clusterID,
+			conf.SegmentMasterKey,
+		); tErr != nil {
+			logrus.WithError(tErr).Errorf("failed to initialize telemetry")
+		} else {
+			DefaultTelemetry = actorDef
+			DefaultTelemetry.syslog.Info(`telemetry reporting is enabled; `,
+				`run with --telemetry-enabled=false to disable`)
+			DefaultTelemetry.telemetryTick(system, 0)
 		}
+	} else {
+		logrus.Info("telemetry reporting is disabled")
+	}
+}
 
-	case telemetryTick:
-		// Tick in a random interval.
-		//nolint:gosec // Weak RNG is fine here.
-		randNum := rand.Intn(maxTickIntervalMins-minTickIntervalMins) + minTickIntervalMins
-		actors.NotifyAfter(ctx, time.Duration(randNum)*time.Minute, telemetryTick{})
+// Track adds track call objects to the analytics.Client interface.
+func (s *TelemetryActor) Track(t analytics.Track) {
+	// Panic if telemetry isn't initialized or has crashed.
+	if s == nil {
+		panic("telemetry actor should not be nil: can't track.")
+	}
 
-		ReportMasterTick(ctx.Self().System(), s.db, s.rm)
+	t.UserId = s.clusterID
+	if err := s.client.Enqueue(t); err != nil {
+		s.syslog.WithError(err).Warnf("failed to enqueue track %s", t.Event)
```

Comment:
I prefer when the wording on the error message explicitly says there's a problem, personally.

Separate topic: since we're doing structured logging, I'd prefer `t.Event` to be a field rather than templated into the message.

```suggestion
		s.syslog.
			WithError(err).
			WithField("event", t.Event).
			Warn("failed to enqueue track")
```

Reply:
i usually take the red line or containing error as 'there is explicitly a problem', but yeah. i'm fine with this. just see a lot of inconsistency "failed to.. ", "error doing..", "doing.."